From 0b6df8a0511e58ac874ef6070e67d62665e8b9cc Mon Sep 17 00:00:00 2001
From: Chanh Nguyen <chanh@os.amperecomputing.com>
Date: Fri, 17 May 2024 14:44:11 +0700
Subject: [PATCH 1/4] spi-nor: check 4b opcode support

Add check if the SPI NOR device supports 4-byte address mode before
setting.
This code is available in upstream u-boot but not sure why it was
removed in OpenBMC u-boot.

Upstream-Status: Pending

Signed-off-by: Chanh Nguyen <chanh@os.amperecomputing.com>
---
 drivers/mtd/spi/spi-nor-core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/spi/spi-nor-core.c b/drivers/mtd/spi/spi-nor-core.c
index 2ccf37d9c8..877645e64c 100644
--- a/drivers/mtd/spi/spi-nor-core.c
+++ b/drivers/mtd/spi/spi-nor-core.c
@@ -2764,7 +2764,8 @@ static int spi_nor_init(struct spi_nor *nor)
 	}

 	if (nor->addr_width == 4 &&
-	    (JEDEC_MFR(nor->info) != SNOR_MFR_SPANSION)) {
+	    (JEDEC_MFR(nor->info) != SNOR_MFR_SPANSION) &&
+	    !(nor->info->flags & SPI_NOR_4B_OPCODES)) {

 		/*
 		 * If the RESET# pin isn't hooked up properly, or the system
--
2.34.1

